bedrock/firefox/tests/test_firefox_all.py

```diff
@@ -0,0 +1,315 @@
+    doc = pq(resp.content)
+
+    link = doc(".c-download-button")
+    assert len(link) == 1
+    download_url = link.attr("href")
+    if "msi" in os:
+        product = "firefox-devedition-msi-latest-ssl"
+        os = os.replace("-msi", "")
+    else:
+        product = "firefox-devedition-latest-ssl"
+    assert all(substr in download_url for substr in [f"product={product}", f"os={os}", f"lang={lang}"])
+    if os.startswith("linux"):
+        linux_link = doc(".c-linux-debian a")
+        assert len(linux_link) == 1
+        assert "https://support.mozilla.org/kb/install-firefox-linux" in linux_link.attr("href")
+
+
+@pytest.mark.parametrize("os, lang", OS_LANG_PAIRS)
+def test_firefox_nightly(client, os, lang):
+    resp = client.get(reverse("firefox.all.download", kwargs={"product_slug": "desktop-nightly", "platform": os, "locale": lang}))
+    doc = pq(resp.content)
+
+    link = doc(".c-download-button")
+    assert len(link) == 1
+    download_url = link.attr("href")
+    if "msi" in os:
+        product = "firefox-nightly-msi-latest-l10n-ssl"
+        os = os.replace("-msi", "")
+    else:
+        product = "firefox-nightly-latest-l10n-ssl"
+    if lang == "en-US":
+        # en-us downloads don't get the l10n releases
+        product = product.replace("-l10n", "")
+    assert all(substr in download_url for substr in [f"product={product}", f"os={os}", f"lang={lang}"])
+    if os.startswith("linux"):
+        linux_link = doc(".c-linux-debian a")
+        assert len(linux_link) == 1
+        assert "https://support.mozilla.org/kb/install-firefox-linux" in linux_link.attr("href")
+
+
+@pytest.mark.parametrize("os, lang", [("linux64-aarch64", "es-ES"), ("linux64-aarch64", "pt-BR")])
+def test_firefox_linux_nightly_aarch(client, os, lang):
+    resp = client.get(reverse("firefox.all.download", kwargs={"product_slug": "desktop-nightly", "platform": os, "locale": lang}))
+    doc = pq(resp.content)
+
+    link = doc(".c-download-button")
+    assert len(link) == 1
+    download_url = link.attr("href")
+    if "msi" in os:
+        product = "firefox-nightly-msi-latest-l10n-ssl"
+        os = os.replace("-msi", "")
+    else:
+        product = "firefox-nightly-latest-l10n-ssl"
+    if lang == "en-US":
+        # en-us downloads don't get the l10n releases
+        product = product.replace("-l10n", "")
+    assert all(substr in download_url for substr in [f"product={product}", f"os={os}", f"lang={lang}"])
+    linux_link = doc(".c-linux-debian a")
+    assert len(linux_link) == 1
+    assert "https://support.mozilla.org/kb/install-firefox-linux" in linux_link.attr("href")
+
+
+@pytest.mark.parametrize("os, lang", OS_LANG_PAIRS)
+def test_firefox_esr(client, os, lang):
+    resp = client.get(reverse("firefox.all.download", kwargs={"product_slug": "desktop-esr", "platform": os, "locale": lang}))
+    doc = pq(resp.content)
+
+    link = doc(".c-download-button")
+    assert len(link) == 1
```

Comment:
I know these tests run against a static copy of product-details (not sure if there's both esr-lastest and esr-next pointing to different versions at that v50.x point in time), but IRL there may be two instances of the button…

Reply:
Ah yes, we should add a test for that. I just need to figure out how to add the `esr-next` in the test. I'll take a look.

Reply:
There's now a test for esr-next and 2 links. Patching the product details was a bit ugly.